src/libstd/ffi/c_str.rs

```diff
@@ -718,6 +719,81 @@ impl From<Box<CStr>> for CString {
     }
 }
 
+#[stable(feature = "cstring_from_vec_of_nonzerou8", since = "1.39.0")]
+impl From<Vec<NonZeroU8>> for CString {
+    /// Converts a [`Vec`]`<`[`NonZeroU8`]`>` into a [`CString`] without
+    /// copying nor checking for inner null bytes.
+    ///
+    /// [`CString`]: ../ffi/struct.CString.html
+    /// [`NonZeroU8`]: ../num/struct.NonZeroU8.html
+    /// [`Vec`]: ../vec/struct.Vec.html
+    #[inline]
+    fn from(v: Vec<NonZeroU8>) -> CString {
+        /// # Safety
+        ///
+        ///   - for any `length: usize`, it must be sound to transmute between
+        ///     `[T; length]` and `[U; length]`.
+        ///
+        ///   - for any `cap: usize`, `Layout<[T; cap]>` needs to be equal to
+        ///     `Layout<[U; cap]>` (for the allocator)
+        #[inline]
+        unsafe
```

Comment:
what's with the line break here?

Reply:
Woops, my own Rust style leaked here (Out of topic: I see `pub` and `unsafe`, among others, as _markers_, hence the logic of them being on an extra line rather than on the same line).